src/controller.cpp

```diff
@@ -46,6 +46,20 @@ tf2::Transform to_transform(const geometry_msgs::Pose & pose)
   return result;
 }
 
+// Returns the square distance between two points
+double distSquared(const tf2::Transform & a, const tf2::Transform & b)
+{
+  return std::pow(a.getOrigin().x() - b.getOrigin().x(), 2) +
+         std::pow(a.getOrigin().y() - b.getOrigin().y(), 2);
+}
+
+// Return the square distance between two points.
+double distSquared(const geometry_msgs::Pose & a, const geometry_msgs::Pose & b)
+{
+  return std::pow(a.position.x - b.position.x, 2) +
+         std::pow(a.position.y - b.position.y, 2);
```

Comment:
I think our internal representation should use `tf2` datatypes because you can perform calculations with them.

Reply:
Btw, `pow` is very very heavy in unoptimized code. A self multiplication would be much faster.

Reply:
Point taken, moving these helper functions improves code quality.